aws/src/test/java/nl/nn/adapterframework/filesystem/AmazonS3FileSystemTestHelper.java

```diff
@@ -0,0 +1,203 @@
+		} else {
+			s3ClientBuilder.withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(serviceEndpoint, clientRegion.getName()));
+		}
+
+		return s3ClientBuilder.build();
+	}
+
+	@Override
+	public void tearDown() throws Exception {
+		if(s3Mock != null) {
+			s3Mock.shutdown();
+		}
+	}
+
+	@Override
+	public boolean _fileExists(String folder, String filename) {
+		String objectName;
+		if(filename == null) {
+			objectName = folder;
+		}else {
+			objectName = folder == null ? filename : folder +"/"+ filename;
+		}
+		try {
+			return s3Client.doesObjectExist(bucketName, objectName);
+		} catch(AmazonS3Exception e) {
+			return false;
+		}
+	}
+
+	@Override
+	public void _deleteFile(String folder, String filename) {
+		String filePath = folder == null ? filename : folder +"/" + filename;
+		s3Client.deleteObject(bucketName, filePath);
+	}
+
+	@Override
+	public OutputStream _createFile(final String foldername, final String filename) throws IOException {
+
+		String fileName = tempFolder.toAbsolutePath()+"tempFile";
+
+		final File file = new File(fileName);
+		final FileOutputStream fos = new FileOutputStream(file);
+		final BufferedOutputStream bos = new BufferedOutputStream(fos);
+
+		FilterOutputStream filterOutputStream = new FilterOutputStream(bos) {
+			@Override
+			public void close() throws IOException {
+				super.close();
+				bos.close();
+
+				FileInputStream fis = new FileInputStream(file);
+				ObjectMetadata metaData = new ObjectMetadata();
+				metaData.setContentLength(file.length());
+				String filePath = foldername == null ? filename : foldername + "/" + filename;
+				s3Client.putObject(bucketName, filePath, fis, metaData);
+
+				fis.close();
+				file.delete();
+			}
+		};
+		return filterOutputStream;
+	}
+
+	@Override
+	public InputStream _readFile(String folder, String filename) throws FileNotFoundException {
+		String path = StringUtil.concatStrings(folder, "/", filename);
+		final S3Object file = s3Client.getObject(bucketName, path);
+		InputStream is = file.getObjectContent();
+		FilterInputStream fos = new FilterInputStream(is) {
+			@Override
+			public void close() throws IOException {
+				super.close();
+				file.close();
+			}
+		};
+
+		return fos;
+	}
+
+	@Override
+	public void _createFolder(String folderName) throws IOException {
+		String foldername = folderName.endsWith("/") ? folderName : folderName +"/";
+		s3Client.putObject(bucketName, foldername, "");
+	}
+
+	@Override
+	public boolean _folderExists(String folderName) throws Exception {
+		String foldername = folderName.endsWith("/") ? folderName : folderName + "/";
+		return _fileExists(foldername, null);
+	}
+
+	@Override
+	public void _deleteFolder(String folderName) throws Exception {
+		String folder = null;
+		if(folderName != null) {
+			folder = folderName.endsWith("/") ? folderName : folderName + "/";
+		}
+		cleanUpFolder(folder);
+	}
+
+	private void cleanUpFolder(String foldername) {
```

Comment:
Ook hier dupliceert onze test-code weer voor een groot deel de code die we aan het testen zijn. :-( 

(Ja ik weet dat dit een move is van een bestaande file dus bestaande code).